map/bookmark_manager.cpp

```diff
@@ -18,16 +18,27 @@
 
 #include <algorithm>
 
+using SearchUserMarkContainer = SpecifiedUserMarkContainer<SearchMarkPoint, UserMark::Type::SEARCH>;
+using ApiUserMarkContainer = SpecifiedUserMarkContainer<ApiMarkPoint, UserMark::Type::API>;
+using DebugUserMarkContainer = SpecifiedUserMarkContainer<DebugMarkPoint, UserMark::Type::DEBUG_MARK>;
+using RouteUserMarkContainer = SpecifiedUserMarkContainer<RouteMarkPoint, UserMark::Type::ROUTING>;
+using LocalAdsMarkContainer = SpecifiedUserMarkContainer<LocalAdsMark, UserMark::Type::LOCAL_ADS>;
+using StaticUserMarkContainer = SpecifiedUserMarkContainer<SearchMarkPoint, UserMark::Type::STATIC>;
+
 BookmarkManager::BookmarkManager(Framework & f)
   : m_framework(f)
 {
-  m_userMarkLayers.reserve(5);
-  m_userMarkLayers.emplace_back(new SearchUserMarkContainer(0.0 /* activePinDepth */, m_framework));
-  m_userMarkLayers.emplace_back(new ApiUserMarkContainer(0.0 /* activePinDepth */, m_framework));
-  m_userMarkLayers.emplace_back(new DebugUserMarkContainer(0.0 /* debugDepth */, m_framework));
-  m_userMarkLayers.emplace_back(new RouteUserMarkContainer(0.0 /* activePinDepth */, m_framework));
-  m_userMarkLayers.emplace_back(new LocalAdsMarkContainer(0.0 /* activePinDepth */, m_framework));
-  UserMarkContainer::InitStaticMarks(FindUserMarksContainer(UserMarkType::SEARCH_MARK));
+  m_userMarkLayers.reserve(6);
+  m_userMarkLayers.emplace_back(new SearchUserMarkContainer(m_framework));
+  m_userMarkLayers.emplace_back(new ApiUserMarkContainer(m_framework));
+  m_userMarkLayers.emplace_back(new DebugUserMarkContainer(m_framework));
+  m_userMarkLayers.emplace_back(new RouteUserMarkContainer(m_framework));
+  m_userMarkLayers.emplace_back(new LocalAdsMarkContainer(m_framework));
+
+  auto staticMarksContainer = new StaticUserMarkContainer(m_framework);
+  m_userMarkLayers.emplace_back(staticMarksContainer);
```

Comment:
```c++
auto staticMarksContainer = make_unique<StaticUserMarkContainer>(m_framework);
UserMarkContainer::InitStaticMarks(staticMarksContainer);
m_userMarkLayers.emplace_back(move(staticMarksContainer));
```
Otherwise, uniqueness of the pointer to static marks container is violated.